modules/programs/zsh.nix

```diff
@@ -616,158 +616,160 @@ in
         ++ optional cfg.enableCompletion pkgs.nix-zsh-completions
         ++ optional cfg.oh-my-zsh.enable cfg.oh-my-zsh.package;
 
-      home.file."${relToDotDir ".zshrc"}".text = concatStringsSep "\n" ([
+      programs.zsh.initExtra = mkMerge [
```

Comment:
actually this will change the results for someone who set `initExtra = lib.mkForce []`, it must be `home.file."${relToDotDir ".zshrc"}".text` for backwards compatibility.

Reply:
I believe this is central to the change, though? A user will pass different priorities to `initExtra` to write to the contents where they want.

Reply:
`home.file."${relToDotDir ".zshrc"}".text = mkBefore ` would still work but yeah that's less visible from the doc. 
Putting all the generated code in `initExtra` feels odd since it's not `extra` anymore ? In a previous version of the PR @Emin017 had added a new `content` option, maybe that's we need after all to keep backwards compatibility.

Reply:
Yeah, I would feel better with a `content` or something that implies its the final config.